### PR TITLE
Ctw 1200/rewire use history

### PR DESCRIPTION
**Pull request by pmadria3**
This PR sets up the drawer to retrieve the versions history for resources (currently conditions) and also rewires the useHistory hook to utilize FQS for resources.

### CODE REVIEW

File: src/components/content/resource/history.ts [hunk 1]
```diff
@@ -29,53 +37,110 @@ export function useHistory<T extends ResourceTypeString, M extends FHIRModel<Res
...
-        return entries;
-      } catch (e) {
-        throw Telemetry.logError(
-          e as Error,
-          `Failed fetching condition history for patient: ${patient.UPID}}`
-        );
-      }
-    }, `req.${model.resourceType.toLowerCase()}_history`),
+    enableFQS
+      ? withTimerMetric(
```
Comment (bryanjknight) [inline src/components/content/resource/history.ts:47]: Let's break these up into two functions; it's a bit hard to read with the inline if-else statement

**Review by bryanjknight — CHANGES_REQUESTED** [review]

File: src/components/content/resource/history.ts [hunk 2]
```diff
@@ -150,3 +147,149 @@ function wasEnteredInError(resource: fhir4.FhirResource) {
...
 }
+
+function getResourceFQSQuery(resourceType: ResourceTypeString) {
+  switch (resourceType) {
+    case "Condition":
+      return conditionsQuery;
+    case "AllergyIntolerance":
+      return allergyQuery;
+    default:
+      return "";
```
Comment (WesleyKapow) [inline src/components/content/resource/history.ts:158]: Let's make sure we don't silently error if we accidentally use this when it isn't setup.
```suggestion
      throw new Error(`Resource type to FQS query not implemented yet for ${resourceType}`);
```

File: src/services/fqs/queries/versions.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    }
+  `;
+}
+
+function getResourceFragment(resourceType: ResourceTypeString) {
+  switch (resourceType) {
+    case "Condition":
+      return fragmentConditionHistory;
+    default:
+      return "";
```
Comment (WesleyKapow) [inline src/services/fqs/queries/versions.ts:25]: Another `throw` here.

File: src/services/fqs/queries/fragments/condition.ts [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+      type {
+        coding {
+          ...Coding
+        }
+      }
+    }
+  }
+`;
+
+export const fragmentConditionHistory = gql`
```
Comment (WesleyKapow) [inline src/services/fqs/queries/fragments/condition.ts:117]: What's the difference between this and the one above? Is there a way to DRY this up?

File: src/components/content/resource/history.ts [hunk 5]
```diff
@@ -150,3 +147,149 @@ function wasEnteredInError(resource: fhir4.FhirResource) {
...
+  }
+}
+
+export async function getVersionHistoryFQS<T extends ResourceTypeString>(
```
Comment (WesleyKapow) [inline src/components/content/resource/history.ts:162]:
```suggestion
async function getVersionHistoryFQS<T extends ResourceTypeString>(
```

**Review by WesleyKapow — CHANGES_REQUESTED** [review]
Looks great, few nitpicks and then it'll be good to go!

File: src/components/content/resource/history.ts [hunk 6]
```diff
@@ -293,3 +306,25 @@ async function fetchResourcesODS<
     );
   }
 }
+
+export function filterLensandSummary<T extends ResourceTypeString>(
```
Comment (WesleyKapow) [inline src/components/content/resource/history.ts:310]: nit: `filterLens**A**ndSummary` (capital A)

File: src/components/content/resource/history.ts [hunk 7]
```diff
@@ -39,7 +43,7 @@ export function useHistory<T extends ResourceTypeString, M extends FHIRModel<Res
   getSearchParams,
   getHistoryEntry,
   getFiltersFQS,
-  enableFQS = true,
+  enableFQS = false,
```
Comment (WesleyKapow) [inline src/components/content/resource/history.ts:46]: nit: Do we need a default here? Do all consumers of this now explicitly pass this?

**Pull request merged**